components/org.wso2.carbon.identity.sso.saml.ui/src/main/java/org/wso2/carbon/identity/sso/saml/ui/client/SamlSPMetadataUploadExecutor.java

```diff
@@ -97,6 +97,12 @@ public boolean execute(HttpServletRequest httpServletRequest, HttpServletRespons
                     attributeConsumingServiceIndex = serviceProviderDTO.getAttributeConsumingServiceIndex();
                 }
 
+                // Store the certificate contained inside the metadata file, in the session.
+                // This will be used by service provider update operation to know the certificate came inside SAML
+                // metadata file.
+                httpServletRequest.getSession().setAttribute("applicationCertificate",
```

Comment:
Move "applicationCertificate" to constant

Reply:
The literal is not a magic literal and is used only once. Do we still need to make it a constant?

Reply:
@rushmin , this is a constant, and putting a value against "applicationCertificate" into session means someone will be using it somewhere. So this needs to be defined as a public constant.